lib/widgets/stream_colors.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_color_models/flutter_color_models.dart';
+
+import '../api/model/model.dart';
+import 'color.dart';
+
+/// A provider for [StreamColorSwatch]es that computes and caches them.
+abstract class StreamColorSwatches {
+  static final StreamColorSwatches light = _StreamColorSwatchesLight._();
+  static final StreamColorSwatches dark = _StreamColorSwatchesDark._();
+
+  final Map<int, StreamColorSwatch> _cache = {};
+
+  /// Gives the cached [StreamColorSwatch] for a [subscription.color].
+  StreamColorSwatch forBaseColor(int base) =>
```

Comment:
```suggestion
  /// Gives the [StreamColorSwatch] for a [subscription.color].
  StreamColorSwatch forBaseColor(int base) =>
```

I wouldn't say this gives "the cached swatch", because there may not have been a cached swatch yet when it was called.